### PR TITLE
get_z_level() proc

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Converts just raw getting thing.z into thing.get_z_leve() which literally just returns z

## Why It's Good For The Game

Useful for NSV overriding.

## Changelog
:cl:
code: Replaces .z with .get_z_level()
/:cl:

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -1248,3 +1248,17 @@
...
+  */
+/atom/proc/get_z_level()
+	if(z)
+		return z
+	else
+		var/turf/T = get_turf(src)
+		if(T)
+			return T.z
+		else
+			return 0
```
Comment (ike709) [inline code/game/atoms.dm:1264]: Not null?

**Comment by ike709** [comment]
What's an actual usecase for this that justifies the proc call overhead? And why does it need to be applied to every z check?

**Comment by PowerfulBacon** [comment]
Allowing to have multiple ships on 1 z-level but treat them as different z-levels. This allows for multiple ships with gravity to be instanced on 1 z-level but with localized gravity, suit sensors etc.
Could change it to a define to remove the proc call overhead.

**Comment by ike709** [comment]
Fair enough, but yeah make it a define.

**Pull request closed without merging**